crates/kornia-io/src/functional.rs

```diff
@@ -137,13 +178,16 @@ pub fn read_image_any(file_path: impl AsRef<Path>) -> Result<Image<u8, 3>, IoErr
 mod tests {
     use crate::error::IoError;
     use crate::functional::read_image_any;
+    use kornia_image::Image;
 
     #[cfg(feature = "jpegturbo")]
     use crate::functional::{read_image_jpeg, write_image_jpeg};
 
     #[test]
     fn read_any() -> Result<(), IoError> {
-        let image = read_image_any("../../tests/data/dog.jpeg")?;
+        // let image: Image<u8, 3> = read_image_any("../../tests/data/dog.jpeg")?;
+        let image: super::GenericImage = read_image_any("../../tests/data/dog.jpeg")?;
+        // NOTE: then how to access the size? we need to reimplment the methods ??
```

Comment:
How about returning a `Box<dyn AnyImage>` with 

```rust
trait AnyImage {
    fn size(&self) -> ImageSize;
}

impl<T: SafeTensorType, const C: usize> AnyImage for Image<T, C> {
    fn size(&self) -> ImageSize {
        self.size()
    }
}

```

Reply:
it's another option, however wouldn't be difficult to scale this as soon as we increase the functionality in Image ?

Reply:
I don't think it would be more effort than the current enum approach but I can give it more thought. What if the AnyImage trait owns all the image functionality? Then we don't have to implement it twice

Reply:
I also think using traits to decouple the functionality may be a good direction, although I haven't fully thought it through.

If considering/implementing the above needs more time, another limited option for now, is to use the enum type only for returning and force the user to convert the enum to a concrete type before further operations. We can provide functions like `into_rgb8() -> Result<Image<u8, 3>, XXError>` etc. in `impl GenericImage {}`.

**Based on use cases:**

- **User who does not care about the pixel type at all**
    Not easily possible with this limited solution. User has to either handle all image types with match (maybe macros?) or convert to one specific type (as below). 

    Trait based solutions may work for this use case.

- **User who cares only about the final image type**
  ```rust
  let gimg: super::GenericImage = read_image_any("../../tests/data/dog.jpeg")?;
  let gray = gimg.into_l8()?;
  // Any operation for Image<u8, 1>
  ```

- **User who wants to handle only specific type images**
  ```rust
  let gimg: super::GenericImage = read_image_any("../../tests/data/dog.jpeg")?;
  if let GenericImage::Rgb8(image) = gimg {
     // Any operation for Image<u8, 3>
  }
  ```

Reply:
that sounds a good option. Also in this direction, i want start getting rid of `image-rs` and implement directly the [decoders](https://github.com/image-rs/image/blob/main/Cargo.toml#L43-L57) we want, which can lead to have a pool of free functions to read from different formats in the format of `read_rgb8` , `read_l8`, etc. Possibly by curating first `png`, jpeg`, `exr` and maybe `tiff` which are the most common formats used in ml/robotics